src/main/java/school/lemon/changerequest/java/container/ContainerImpl.java

```diff
@@ -1,38 +1,106 @@
 package school.lemon.changerequest.java.container;
 
-public class ContainerImpl implements Container{
+import java.util.Arrays;
+
+public class ContainerImpl implements Container {
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+    private int size;
+    transient int[] elementData;
+
+    public ContainerImpl() {
+        this.elementData = new int[INITIAL_ARRAY_SIZE];
+    }
 
     @Override
     public int size() {
-        // TODO: please implement me
-        return 0;
+        return size;
     }
 
     @Override
     public void clear() {
-        // TODO: please implement me
+        // clear to let GC do its work
+        for (int i = 0; i < size; i++)
+            elementData[i] = 0;
+
+        size = 0;
     }
 
     @Override
     public Integer get(int index) {
-        // TODO: please implement me
-        return null;
+        if (index >= size){
+            return null;
+        }
+        return elementData[index];
     }
 
     @Override
     public void add(int element) {
-        // TODO: please implement me
+        ensureCapacityInternal(size + 1);
+        elementData[size++] = element;
     }
 
     @Override
     public boolean add(int element, int index) {
-        // TODO: please implement me
+        if (index <= size && index >= 0) {
+            ensureCapacityInternal(size + 1);
+            System.arraycopy(elementData, index, elementData, index + 1,
+                    size - index);
+            elementData[index] = element;
+            size++;
+            return true;
+        }
         return false;
     }
 
     @Override
     public boolean remove(int index) {
-        // TODO: please implement me
+        for (int i = 0; i < size; i++)
+            if (index == i) {
+                fastRemove(i);
+                return true;
+            }
         return false;
     }
+
+    private void fastRemove(int index) {
+        int numMoved = size - index - 1;
+        if (numMoved > 0)
+            System.arraycopy(elementData, index + 1, elementData, index,
+                    numMoved);
+        elementData[--size] = 0;
+    }
+
+    private void ensureCapacityInternal(int minCapacity) {
+        if (elementData.length == INITIAL_ARRAY_SIZE) {
+            minCapacity = Math.max(INITIAL_ARRAY_SIZE, minCapacity);
+        }
+
+        ensureExplicitCapacity(minCapacity);
+    }
+
+    private void ensureExplicitCapacity(int minCapacity) {
+        // overflow-conscious code
+        if (minCapacity - elementData.length > 0)
+            grow(minCapacity);
+    }
+
+    private void grow(int minCapacity) {
```

Comment:
It's a hard to understand what you are doing here. Please add comments.
By the way, looks like copy-paste, so not sure you are aware what is going on.